src/utils.jl

```diff
@@ -0,0 +1,16 @@
+"""
+    submat(A::AbstractPDMat, inds)
+
+Select a square submatrix of an `AbstractPDMat` specified by `inds`.
+Returns a `AbstractPDMat` of the same type.
+"""
+submat(A::PDMat, inds) = PDMat(A[inds, inds])
+submat(A::PSDMat, inds) = PSDMat(A[inds, inds])
```

Comment:
just curious, is a submat of PSDMat guaranteed to be PSDMat? I would think so, but I remember when I implemented this with distributions, it conplains things not being Hermitian when wrapping it into a distribution. Only happens to `PSDMat`.

Reply:
>just curious, is a submat of PSDMat guaranteed to be PSDMat? 

It should be, at least for real-valued matrices I don't see why it wouldn't. If/when you come across this again maybe open an issue here.